### PR TITLE
Added unit tests to the functions that writes group & field data to Eclipse-restart file

### PR DESCRIPTION
**Pull request by jalvestad**
Added unit tests to ensure that the code works as they should on the  group structure in the
example data set. The structure contains two group levels, a well group level and a node group
level.

### CODE REVIEW

File: tests/test_AggregateGroupData.cpp [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+	state.add("FWPR",   5678.);
+
+	return state;
+    }
+}
+
+struct SimulationCase
+{
+    explicit SimulationCase(const Opm::Deck& deck)
+        : es   { deck }
```
Comment (joakim-hove) [inline tests/test_AggregateGroupData.cpp]: Why do you use `{ .. }` instead of plain `es(deck)`?

File: tests/test_AggregateGroupData.cpp [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+#include <utility>
+#include <vector>
+#include <iostream>
+#include <cstddef>
+
+struct MockIH
+{
+    MockIH(const int numWells,
+	   
+	   const int igrpPerGrp	 = 101,  // no of data elements per group in IGRP array
```
Comment (joakim-hove) [inline tests/test_AggregateGroupData.cpp]: These numbers - are they specific to this testcase?

File: tests/test_AggregateGroupData.cpp [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+    this->nsgrpz = this->value[Ix::NSGRPZ] = sgrpPerGrp;
+    this->nxgrpz = this->value[Ix::NXGRPZ] = xgrpPerGrp;
+    this->nzgrpz = this->value[Ix::NZGRPZ] = zgrpPerGrp;
+}
+
+namespace {
+    Opm::Deck first_sim()
+    {
+        // Mostly copy of tests/FIRST_SIM.DATA
+        const auto input = std::string {
```
Comment (joakim-hove) [inline tests/test_AggregateGroupData.cpp]: The type must be *somewhere* - why nut just:
```
const std::string input = R"( ... )";
```

**Comment by joakim-hove** [comment]
jenkins build this please

File: tests/test_AggregateGroupData.cpp (hunk 2 above)
Comment (jalvestad) [reply tests/test_AggregateGroupData.cpp]: No - the numbers that are assigned constant values, always has these values. I.e. igrpPerGrp, sgrpPerGrp, xgrpPerGrp and zgrpPerGrp.

File: tests/test_AggregateGroupData.cpp (hunk 3 above)
Comment (jalvestad) [reply tests/test_AggregateGroupData.cpp]: OK - Changed.

File: tests/test_AggregateGroupData.cpp (hunk 2 above)
Comment (joakim-hove) [reply tests/test_AggregateGroupData.cpp]: OK - sorry for being a bit dense; what I am after is just that *iff* these numbers are truly constant they should be defined as named constants in the output code, and not inline in the test code. But maybe I misunderstand.

**Comment by jalvestad** [comment]
Added a correction to the code for a compilation error (check on string-variables). Adjusted code to account for comments from Joakim. Jostein

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by jalvestad** [comment]
I consider this development as finished and the code is ready for final review and merging.

**Comment by bska** [comment]
jenkins build this please

File: tests/test_AggregateGroupData.cpp [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+
+
+END
+
+)~" };
+
+        return Opm::Parser{}.parseString(input);
+    }
+
+        Opm::SummaryState sim_state()
```
Comment (joakim-hove) [inline tests/test_AggregateGroupData.cpp]: This seems to be unused?

File: tests/test_AggregateGroupData.cpp [hunk 5]
```diff
@@ -0,0 +1,589 @@
...
+}
+
+struct SimulationCase
+{
+    explicit SimulationCase(const Opm::Deck& deck)
+        : es   ( deck )
+        , sched (deck, es )
+    {}
+
+    // Order requirement: 'es' must be declared/initialised before 'sched'.
```
Comment (joakim-hove) [inline tests/test_AggregateGroupData.cpp]: Uninteresting comment.

**Pull request merged**